code/mob/living/silicon/ai.dm

```diff
@@ -1886,6 +1886,38 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 	else
 		boutput(usr, "<span class='alert'><b>Radio not found!</b></span>")
 
+/mob/living/silicon/ai/verb/open_map()
+	set name = "Open station map"
+	set desc = "Click on the map to teleport"
+	set category = "AI Commands"
+
+	var/datum/tgui/map_ui
+	var/mob/message_mob = src.get_message_mob()
+	if (!src || !message_mob.client || isdead(src))
+		return
+
+	map_ui = tgui_process.try_update_ui(usr, message_mob, map_ui)
+	if (!map_ui)
+		if (!winexists(usr, "ai_map"))
+			winset(message_mob.client, "ai_map", list2params(list(
+				"type" = "map",
+				"size" = "300,300",
+			)))
+			var/atom/movable/screen/handler = new
+			handler.plane = 0
+			handler.mouse_opacity = 0
+			handler.screen_loc = "ai_map:1,1"
+			message_mob.client.screen += handler
+
+			ai_station_map.screen_loc = "ai_map;1,1"
+			handler.vis_contents += ai_station_map
+			message_mob.client.screen += ai_station_map
+		map_ui = new(usr, message_mob, "AIMap")
+		map_ui.open()
+
+	if (!src.deployed_to_eyecam)
```

Comment:
It would probably be better to have the eye eject only when you click to teleport on the map, rather than when you open it, see `station_map.dm` for the click handler

Reply:
It looks like the click handler wants to work for all mobs, so the naive approach of moving `deployed_to_eyecam`/`eye_view()` over to the handler fails to build as they're undefined for a generic mob.

Is there a proper way to use/notify/signal the mainframe fields/procs from the click handler?

Reply:
You could just do
```
var/mob/living/silicon/ai/mainframe = usr
if (istype(mainframe))
	mainframe.map_teleport(x,y)
```
Where `map_teleport` pops the eyecam out and sets its location